sqlite/sqlite.go

```diff
@@ -0,0 +1,144 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strconv"
+	"sync"
+
+	// sqlite3 driver
+	_ "github.com/mattn/go-sqlite3"
+
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultFilename = "influxd.sqlite"
+	InmemPath       = ":memory:"
+)
+
+// SqlStore is a wrapper around the db and provides basic functionality for maintaining the db
+// including flushing the data from the db during end-to-end testing.
+type SqlStore struct {
+	mu  sync.Mutex
+	db  *sql.DB
+	log *zap.Logger
+}
+
+func NewSqlStore(path string, log *zap.Logger) (*SqlStore, error) {
+	db, err := sql.Open("sqlite3", path)
+	if err != nil {
+		return nil, err
+	}
+	log.Info("Resources opened", zap.String("path", path))
+
+	// If using an in-memory database, don't allow more than 1 connection. Each connection
+	// is given a "new" database. We can't use a shared cache in-memory database because
+	// parallel tests that run multiple launchers in the same process will have issues doing
+	// concurrent writes to the database. See: https://sqlite.org/inmemorydb.html
+	if path == InmemPath {
+		db.SetMaxOpenConns(1)
+	}
+
+	return &SqlStore{
+		db:  db,
+		log: log,
+	}, nil
+}
+
+// Close the connection to the sqlite database
+func (s *SqlStore) Close() error {
+	err := s.db.Close()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Flush deletes all records for all tables in the database.
+func (s *SqlStore) Flush(ctx context.Context) {
+	tables, err := s.tableNames()
+	if err != nil {
+		s.log.Fatal("unable to flush sqlite", zap.Error(err))
+	}
+
+	for _, t := range tables {
+		stmt := fmt.Sprintf("DELETE FROM %s", t)
+		err := s.execTrans(stmt)
+		if err != nil {
+			s.log.Fatal("unable to flush sqlite", zap.Error(err))
+		}
+	}
+	s.log.Debug("sqlite data flushed successfully")
+}
+
+func (s *SqlStore) execTrans(stmt string) error {
```

Comment:
We will probably end up using something like `upper/db` for the application code in the feature-aligned services to interact with metadata resources stored in the database to be consistent with how the cloud services are being set up. This `execTrans` method is used for the more base-level actions like doing migrations and flushing the db.